### PR TITLE
CDAP-15854: Changed packages and updated pom.

### PR DESCRIPTION
**Pull request by AntonGensitskiy**
- Update packages
- Make some pom changes

https://issues.cask.co/browse/CDAP-15854

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -18,25 +18,22 @@
...
   <name>Data Pipeline Batch Plugins</name>
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <spark.version>1.6.1</spark.version>
-    <cdap.version>4.1.0</cdap.version>
+    <cdap.version>6.1.0-SNAPSHOT</cdap.version>
     <hadoop.version>2.3.0</hadoop.version>
-    <hydrator.plugins.version>1.6.0</hydrator.plugins.version>
+    <hydrator.plugins.version>2.2.0-SNAPSHOT</hydrator.plugins.version>
```
Comment (aonischuk) [inline pom.xml]: 2.3.0-SNAPSHOT I think is the latest.

File: pom.xml [hunk 2]
```diff
@@ -18,25 +18,22 @@
...
     <!-- properties for script build step that creates the config files for the artifacts -->
     <widgets.dir>widgets</widgets.dir>
```
Comment (aonischuk) [inline pom.xml]: remove

File: pom.xml [hunk 3]
```diff
@@ -18,25 +18,22 @@
...
     <docs.dir>docs</docs.dir>
```
Comment (aonischuk) [inline pom.xml]: same

File: pom.xml [hunk 4]
```diff
@@ -282,10 +279,10 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <version>3.1</version>
+        <version>3.7.0</version>
```
Comment (aonischuk) [inline pom.xml:279]: You forgot to change
`<_exportcontents>co.cask.hydrator.*</_exportcontents>`

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,36 @@
+*.class
```
Comment (aonischuk) [inline .gitignore:1]: Please add checkstyle and rat plugins.
Also suppressions.xml and checkstyle.xml files.

**Review by aonischuk — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 5 above)
Comment (AntonGensitskiy) [reply .gitignore:1]: Done.

File: pom.xml (hunk 4 above)
Comment (AntonGensitskiy) [reply pom.xml:279]: Done.

File: pom.xml (hunk 3 above)
Comment (AntonGensitskiy) [reply pom.xml]: Done.

File: pom.xml (hunk 2 above)
Comment (AntonGensitskiy) [reply pom.xml]: Done.

File: pom.xml (hunk 1 above)
Comment (AntonGensitskiy) [reply pom.xml]: Done.

**Pull request merged**